src/cli/client_manager.py

```diff
@@ -119,6 +119,48 @@ def check_pkh_known_by_signer(self, key_name, timeout=None):
                 raise ClientException(f'The secret key of the payout address {key_name} was not imported to the signer!\n'
                                       f'{signer_exception}')
 
+    def baker_exists(self, address, timeout=None):
+        """A baker exists if the public key was revealed."""
+        url = "{}/chains/main/blocks/head/context/contracts/{}/manager_key".format(self.node_endpoint, address)
+        baker_address_exception = f'Error querying the baker at url {self.node_endpoint}.'
+        try:
+            response = self._do_request(method="GET",
+                                        url=url,
+                                        timeout=timeout)
+        except Exception as e:
+            raise ClientException(f'Exception: {e}\n{baker_address_exception}')
+
+        if response.status_code != HTTPStatus.OK:
+            logger.error(f'{response.text}\n{baker_address_exception}')
+            return False
+
+        response_json = response.json()
+        if response_json is None or len(response_json) != 54 or not response_json.startswith("edpk"):
+            logger.error(f'No public key found for {address}, which means a revelation operation has not been published.')
+            return False
+        return True
+
+    def baker_delegatable(self, address, timeout=None):
+        """A baker is delegatable if the response has a delegate key and the baker is delegating to itself."""
+        url = "{}/chains/main/blocks/head/context/contracts/{}".format(self.node_endpoint, address)
+        baker_address_exception = f'Error querying the baker at url {self.node_endpoint}.'
+        try:
+            response = self._do_request(method="GET",
+                                        url=url,
+                                        timeout=timeout)
+        except Exception as e:
+            raise ClientException(f'Exception: {e}\n{baker_address_exception}')
+
+        if response.status_code != HTTPStatus.OK:
+            logger.error(f'{response.text}\n{baker_address_exception}')
+            return False
+
+        response_json = response.json()
+        if response_json is None or "delegate" not in response_json or response_json["delegate"] != address:
```

Comment:
Please correct me if it's a bad way to detect if an address is delegatable.

Reply:
What if the endpoint returns 500? Is it correct to assume false for anything but 200?

RpcBlockApiImpl has another revelation operation. We better remove one of them. Also please correct me if I am wrong, the payment address also needs to be revealed but currently it is not checked. @amzid please comment.

Reply:
@habanoz thanks for mentioning the `RpcBlockApiImpl` Im still familiarizing myself with the codebase. I moved the validation from the client_manager into the yaml_conf_parser without any status code checking.

As I understand it correctly a reveal operation is not mandatory when you want to send tezos to this payment address. Thus a reveal is not a validator for a payment address, or?

Reply:
The payee does not need to be revealed. However, the payer may need to be revealed. However, I am not sure. 

https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/issues/11

I found a past issue that mentions revelation need.

Reply:
@habanoz ok I will open a seperate PR for the payment address revelation check